coreos-installer

```diff
@@ -664,31 +682,87 @@ log() {
     echo "$1" >&2
 }
 
+#########################################################
+#Update bootloader if needed
+#########################################################
+write_bootloader() {
+    if [[ "$arch" == "s390x" ]]; then
+        log "Updating zipl"
+        mkdir -p /mnt/boot_partition
+        mount_boot_partition /mnt/boot_partition
+        trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+
+        blsfile=$(ls /mnt/boot_partition/loader/entries/*.conf)
+        echo "$(grep options $blsfile | cut -d' ' -f2-) $(cat /tmp/s390x_opts) ignition.firstboot rd.neednet=1 ip=dhcp" > /tmp/zipl_prm
+        zipl --verbose -p /tmp/zipl_prm -i $(ls /mnt/boot_partition/ostree/*/*vmlinuz*) -r $(ls /mnt/boot_partition/ostree/*/*initramfs*) --target /mnt/boot_partition
+        if [[ $? -ne 0 ]]; then
+            log "failed updating bootloder"
+            exit 1
+        fi
+        # this allows a reboot into new installed disk
+        chreipl "${DEST_DEV}"
+    fi
+}
+#########################################################
+#Handle s390x zFCP SCSI devices
+#########################################################
+write_image_to_zfcp_disk() {
+    log "Extracting disk image"
+    mount_tmpfs $(( $($DECOMPRESSION_TOOL ${imagefile_compressed} | wc --bytes) / 1024 / 1024 + ${TMPFS_MBSIZE} ))
+    $DECOMPRESSION_TOOL ${imagefile_compressed} > ${imagefile}
+
+    (while IFS='' read -r line || [ -n "$line" ]
+    do
+        if [[ $line =~ "start=" ]]
+        then
+            # we save the start= and size= here, then dd the partitions later
+            printf "%s:%s\n" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 1 | tr -d ' ')" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 2 | tr -d ' ')" \
+                >> /tmp/sfdisk_info
```

Comment:
Why write these to a temp file that will have to be parsed back in rather than just storing them inside of an array?

Reply:
Good point. I'm redoing this.

Reply:
Ah now I remember why I did this: because this while-do is executed inside a block `( )` so any variable inside that chunk will not be available outside that block.

Reply:
The variable could be created at a higher scope, and then used, but I follow. It may be a good idea to add a comment to that effect. I could see folks looking at this in 3-6 months and getting a little tripped up :slightly_smiling_face:.

Reply:
@ashcrow : but iirc any change to the variable inside the block would not be remembered outside. Adding a comment.

Reply:
Hmm, is there a reason that the `dd` couldn't just be done inside of this loop?

Reply:
@arithx : The reason is : output of this loop will look like this :

Reply:
Oh, whoops, missed the `sfdisk` pipe at the end.

Reply:
Feel free to resolve or add more comments, many thanks.